### PR TITLE
Allow topnav items added by packages

### LINKED ISSUE #3181: Allow topnav items added by packages
**Describe the bug**
If a package adds items to the top nav and the dynamic-ui package is enabled an error is raised

**To Reproduce**
1. Install the translations package
2. Install de dynamic-ui package
3. Create a menu with the dynamic-ui package.
4. Assign to the 'admin' user the menu created in step 3
5. Reload the page.

**Expected behavior**
Item menus added by a package should be added to the ones defined by the dynamic-ui package. 

**Screenshots**
image

### PR DESCRIPTION
**Pull request by danloa**
Resolves #3181 

The layout was modified so that when a topnav provider is registerd (for example the dynami-ui package) items added by packages are rendered. All PM core menu items in the topnav are removed.

### CODE REVIEW

File: resources/views/layouts/navbar.blade.php [hunk 1]
```diff
@@ -20,20 +20,31 @@
...
-
             @if(Menu::get('customtopnav'))
-            @foreach(Menu::get('customtopnav')->items as $item)
-                <li class="nav-item">
-                    <a  target="{{ $item->attributes['target'] }}" href="{{ $item->url() }}" class="nav-link {{ $item->isActive === true ? 'active': ''}} {{ $item->attributes['class_link'] }} " >
-                    {!! $item->title !!}
-                    </a>
-                </li>
-            @endforeach
+                @foreach(Menu::get('customtopnav')->items as $item)
```
Comment (velkymx) [inline resources/views/layouts/navbar.blade.php:24]: Can we move this loop into a VueJS loop and load the elements as an array in the JS?

File: resources/views/layouts/navbar.blade.php [hunk 2]
```diff
@@ -20,20 +20,31 @@
...
         <b-navbar-nav class="d-flex align-items-center">
-            @foreach(Menu::get('topnav')->items as $item)
-                <b-nav-item href="{{ $item->url() }}" {{$item->isActive !== false ? 'active': ''}}>
-                    {!! $item->title !!}
-                </b-nav-item>
-            @endforeach
-
             @if(Menu::get('customtopnav'))
-            @foreach(Menu::get('customtopnav')->items as $item)
-                <li class="nav-item">
```
Comment (velkymx) [inline resources/views/layouts/navbar.blade.php:31]: Can we change this back to use the b-nav-item? https://bootstrap-vue.org/docs/components/navbar#navbar

File: resources/views/layouts/navbar.blade.php [hunk 3]
```diff
@@ -20,20 +20,31 @@
...
+                        {!! $item->title !!}
+                        </a>
+                    </li>
+                @endforeach
+
+                @foreach(Menu::get('topnav')->items as $item)
+                    @php
+                        $itemRoute = Route::getRoutes()->match(Request::create($item->url()))
+                    @endphp
+                    @if(!$itemRoute->isFallBack && strpos($itemRoute->action['controller'], "ProcessMaker\\Http\\") !== 0)
```
Comment (caleeli) [inline resources/views/layouts/navbar.blade.php:36]: Could you use a descriptive variable to clarify this condition:
`strpos($itemRoute->action['controller'], "ProcessMaker\\Http\\") !== 0`

File: resources/views/layouts/navbar.blade.php [hunk 4]
```diff
@@ -20,20 +20,31 @@
...
+                        <b-nav-item href="{{ $item->url() }}" {{$item->isActive !== false ? 'active': ''}}>
```
Comment (caleeli) [inline resources/views/layouts/navbar.blade.php:37]: What is the data type of `$item->isActive`? is it mixed? not boolean? why you need to do a `!==false` comparison?

Comment (danloa) [reply resources/views/layouts/navbar.blade.php:37]: That is the way how ProcessMaker's top menu was built. isActive is boolean and is used by <b-nav-item> to highlight the current selected menu item.

File: resources/views/layouts/navbar.blade.php (hunk 3 above)
Comment (danloa) [reply resources/views/layouts/navbar.blade.php:36]: done

File: resources/views/layouts/navbar.blade.php (hunk 2 above)
Comment (danloa) [reply resources/views/layouts/navbar.blade.php:31]: Fixed, changed to <b-nav-item>

File: resources/views/layouts/navbar.blade.php (hunk 1 above)
Comment (danloa) [reply resources/views/layouts/navbar.blade.php]: Moved to a Vue v-for loop

File: resources/views/layouts/navbar.blade.php [hunk 5]
```diff
@@ -19,23 +19,36 @@
...
+                    return !$isCoreLink || $item['isCustom'];
+                });
+            }
+        @endphp
+
         <b-navbar-nav class="d-flex align-items-center">
-            @foreach(Menu::get('topnav')->items as $item)
-                <b-nav-item href="{{ $item->url() }}" {{$item->isActive !== false ? 'active': ''}}>
-                    {!! $item->title !!}
+                <b-nav-item v-for="item in {{ json_encode ($menuItems) }}"
```
Comment (velkymx) [inline resources/views/layouts/navbar.blade.php:44]: Move the menuItems to a data element in the javascript. The menu items shouldn't be processed on each page load.

Like this:

<script>
export default {
  data() {
    return {
      menuitems: {{ json_encode ($menuItems) }}
    }
  },

Comment (danloa) [reply resources/views/layouts/navbar.blade.php]: The backend logic was moved to a manager and the JS variable is set in the layout.

**Pull request merged**